components/engine_traits/src/metrics_task.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::time::{Duration, Instant};
+
+use tikv_util::IntervalRunnable;
+
+use crate::engine::KvEngine;
+use crate::engines::Engines;
+use crate::raft_engine::RaftEngine;
+
+const FLUSHER_RESET_INTERVAL: Duration = Duration::from_millis(60_000);
+
+pub struct MetricsTask<K: KvEngine, R: RaftEngine> {
```

Comment:
I agree with @brson this does not belong to the engine trait. Maybe move to tikv_util?

Reply:
This can't live in tikv_util because this code depends on code in engine_traits, and engine_traits depends on tikv_util. It could live it tikv directly, or in some other intermediate crate between engine_traits and tikv.